cmd/trust/keyset.go

```diff
@@ -253,3 +302,102 @@ func doListKeysets(ctx *cli.Context) error {
 
 	return nil
 }
+
+func doShowKeyset(ctx *cli.Context) error {
+	if len(ctx.Args()) == 0 {
+		return fmt.Errorf("Please specify keyset name. Select from 'trust keyset list'")
+	}
+
+	keysetName := ctx.Args()[0]
+	if keysetName == "" {
+		return fmt.Errorf("Please specify keyset name. Select from 'trust keyset list'")
+	}
+
+	moskeysetPath, err := getMosKeyPath()
+	if err != nil {
+		return err
+	}
+
+	keysetPath := filepath.Join(moskeysetPath, keysetName)
+	if !PathExists(keysetPath) {
+		return fmt.Errorf("Unknown keyset '%s', cannot find keyset at path: %q", keysetName, keysetPath)
+	}
+
+	// no keyset key name specified only, print path if --path, otherwise list all key dir names
+	if len(ctx.Args()) < 2 {
+		if ctx.Bool("path") {
+			fmt.Printf("%s\n", keysetPath)
+			for _, keyDir := range KeysetKeyDirs {
+				keyPath := filepath.Join(keysetPath, keyDir)
+				fmt.Printf("%s\n", keyPath)
+			}
+		} else {
+
+			if err := tree.PrintDirs(keysetPath, KeysetKeyDirs); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	keyName := ctx.Args()[1]
+	if keyName == "" {
+		return fmt.Errorf("Please specify keyset key name, must be one of: %s", strings.Join(KeysetKeyDirs, ", "))
+	}
+
+	if !isValidKeyDir(keyName) {
+		return fmt.Errorf("Invalid keyset key name '%s':, must be one of: %s", strings.Join(KeysetKeyDirs, ", "))
+	}
+
+	keyPath := filepath.Join(keysetPath, keyName)
+	if !PathExists(keyPath) {
+		return fmt.Errorf("Keyset %s key %q does not exist at %q", keysetName, keyName, keyPath)
+	}
+
+	if len(ctx.Args()) > 2 {
+		item := ctx.Args()[2]
+		fullPath := filepath.Join(keyPath, item)
+		if !PathExists(fullPath) {
+			return fmt.Errorf("Failed reading keyset %s key %s item %s at %q: %w", keysetName, keyName, item, fullPath, err)
+		}
+
+		contents, err := os.ReadFile(fullPath)
+		if err != nil {
+			return fmt.Errorf("Failed reading keyset %s key %s item %s at %q: %w", keysetName, keyName, item, fullPath, err)
+		}
+		if ctx.Bool("path") {
+			fmt.Println(fullPath)
+		} else if ctx.Bool("value") {
+			fmt.Printf("%s", string(contents))
+		} else {
+			fmt.Printf("%s\n%s\n", fullPath, string(contents))
+		}
+		return nil
+	}
+
+	// no item specified, crawl dir and print contents or path
+	keyFiles, err := os.ReadDir(keyPath)
+	if err != nil {
+		return fmt.Errorf("keyset %s key %s directory %q: %w", keysetName, keyName, keyPath, err)
+	}
+
+	printPath := ctx.Bool("path")
+	for _, dEntry := range keyFiles {
+		if dEntry.IsDir() {
+			continue
+		}
+		keyFile := dEntry.Name()
+		fullPath := filepath.Join(keyPath, keyFile)
+		contents, err := ioutil.ReadFile(fullPath)
```

Comment:
Sorry, but you left this one.

Reply:
Sure.  Though I think we need a follow-up PR:

```
$ git-grep ioutil 
cmd/trust/keyset.go:    "io/ioutil"
cmd/trust/keyset.go:            contents, err := ioutil.ReadFile(fullPath)
pkg/trust/hwrng.go:     "io/ioutil"
pkg/trust/hwrng.go:     data, err := ioutil.ReadFile(poolSizePath)
pkg/trust/tpm2-tools.go:        "io/ioutil"
pkg/trust/tpm2-tools.go:        f, err := ioutil.TempFile("/tmp", "pcr")
pkg/trust/tpm2-tools.go:        contents, err := ioutil.ReadFile(name)
pkg/trust/tpm2-tools.go:        f, err := ioutil.TempFile(c.tmpDir, "")
pkg/trust/tpm2.go:      "io/ioutil"
pkg/trust/tpm2.go:      infoBytes, err := ioutil.ReadFile(infoPath)
pkg/trust/tpm2.go:      tmpd, err := ioutil.TempDir("/run", "atx-trustroot-*")
pkg/trust/tpm2.go:      err = ioutil.WriteFile(filepath.Join(dest, "server.crt"), []byte(provCert), 0600)
pkg/trust/tpm2.go:      err = ioutil.WriteFile(filepath.Join(dest, "server.key"), []byte(privKey), 0600)
```

Reply:
Agreed.  Thanks.